src/main.h

```diff
@@ -157,8 +157,12 @@ struct config_t {
   bool api;                     // REST API enabled or not
   char api_key[32];             // API Key
   uint8_t powerup_mode;         // Power Up Mode
+  my92xx_model_t chip_type;          // Device Type
```

Comment:
There is no need to store the device type in the EEPROM. The EEPROM is used only for runtime settings and the device type is something to be determined on compile time only. (i.e. Once compiled for the desired device type, no changes are to be done when the device is running).

Reply:
I purposely set it this way so that all we would need to do is expose this in the UI and then somebody can upload a precompiled version to there bulb- change requisite settings and reboot bulb.

Reply:
I haven't had a chance yet to see if I have the skill set to integrate this into the UI in a setting page.

Reply:
I actually changed how the my92xx Libray was instantiated to allow it to be dynamic instead of explicit at build time.

Reply:
Ah I see. Wouldn't this only benefit people that have bulbs of multiple types? I.e. they have 1 firmware and then on the UI select the one that fits the bulb type? (Just trying to understand the rationale) :)

Reply:
@stelgenhof  It's so people don't need to rebuild the firmware. They can just load the bin you provide in the release. That opens firmware up to a larger selection of people skillwise.

I also plan at some point to make sure the firmware works/with document process of using with  tuyaconvert(https://github.com/ct-Open-Source/tuya-convert) That way people would never need to install arduino suite or vscode and also if using a tuya based bulb never need to even solder anything.(Though tuya-convert does require many steps)

Reply:
Yes multiple firmware could be built but seems to me that having a default firmware and then build options for what those defaults are make sense for people who want to build there own that are one click flash+setup.

I know I have a bad habit of having very mixed IOT devices

Reply:
Thanks! I like this idea of having a ready made firmware that doesn't require any compiling. Perhaps I'll make one version that is install ready (no compilation) and one version for the hackers, tinkerers amongst us. Let me see what needs to be changed for that to work :)

Reply:
@stelgenhof  Don't you already ship a .bin file with the release is that ready to go in AP mode(https://github.com/stelgenhof/AiLight/releases/download/v0.6.0/ailight-0.6.bin). Do you need to change things in that image to make it work like that or did I misunderstand your comment above and your  just looking at what changes you need to expose the chip settings in the web interface for the functionality I added?

Reply:
Yes, there are already binaries with each release. There are a few tidbits on my mind that could go in the binary as well. Also considering a Captive portal so users get redirected to the UI immediately. So nothing related to this PR :)

Reply:
Are we good to merge then as is and then add UI bits later to change settings on rom?